docs/assets/js/main.js

```diff
@@ -36,14 +39,24 @@ if ( tabs && tabs.length > 0 ) {
   }
 }
 
+function handleToggleAllClick( event ) {
+  event.preventDefault();
+  toggleAllDetails( toggleAllBtn );
+}
+
 /**
  * @param {MouseEvent} event - The mouse event object from the click.
  */
-function handleDocumentClick( event ) {
-  const target = event.target;
-  if ( target.matches( `[${ TOGGLE_ATTRIBUTE }]` ) ) {
-    toggleDetails( target );
-  }
+function handleToggleClick( event ) {
+  let target = event.target;
+  toggleDetails( target );
 }
 
-document.addEventListener( 'click', handleDocumentClick, false );
+const toggleAllBtn = document.querySelector( '#toggle-details' );
+const toggleBtns = document.querySelectorAll( '.a-toggle_code button' );
+
+toggleAllBtn.addEventListener( 'click', handleToggleAllClick, false );
+
+for ( let i = 0, len = toggleBtns.length; i < len; i++ ) {
+  toggleBtns[i].addEventListener( 'click', handleToggleClick, false );
```

Comment:
I'm a fan of event delegation by attaching one listener to the document instead of a separate listener for every element but given the simplicity of this website I think either approach is fine.

Reply:
Yeah I did this because if you have the listener on the document, `event.currentTarget` doesn't work and you have to pick out whether the user clicked the button or the SVG icon it contains.